Remove deprecated packages + Lerna

jgroth opened the pull request:
Removing Lerna and the deprecated packages

Should not be merged until #531 has been merged

jgroth commented on .github/workflows/release.yml:
```diff
@@ -1,26 +1,42 @@
-name: Release and Publish
+name: Create and Publish Full Release
```
@adrianschmidt I copied this from lime elements, went over it, and removed stuff that looked like it was specific for that repo. Feel free to correct any mistakes I might have made :smile:

vicgeralds reviewed (commented):
> chore: fix broken lockfile

What did you do to fix it? How was it broken? 😄

jgroth commented:
> > chore: fix broken lockfile
> 
> What did you do to fix it? How was it broken? smile

NPM crashed as soon as I tried to run `npm install` with the message `npm ERR! must provide string spec`. I tried to Google a solution but could not find an answer that worked. It started to happen after a merge conflict.
To solve it I copied the latest version of the lockfile from `main` on GitHub and ran `npm install` again

vicgeralds commented:
> 
> NPM crashed as soon as I tried to run `npm install` with the message `npm ERR! must provide string spec`. I tried to Google a solution but could not find an answer that worked. It started to happen after a merge conflict. To solve it I copied the latest version of the lockfile from `main` on GitHub and ran `npm install` again

Ok, maybe fixup the "chore: remove Lerna" commit instead?

Strange, I didn't have any problem with npm 8.11.0.

vicgeralds commented on tsconfig.json:
```diff
@@ -14,5 +14,6 @@
     },
```
We can remove `paths` here, and `moduleNameMapper` in stencil.config.ts

vicgeralds commented on .eslintrc.js:
```diff
@@ -79,4 +129,36 @@ module.exports = {
+    overrides: [
+        {
+            files: [
+                './**/src/**/*.spec.{ts,tsx}',
+                './**/src/**/*.e2e.{ts,tsx}',
```
```suggestion
                './src/**/*.spec.ts',
```

vicgeralds commented on testing/package.json:
```diff
@@ -2,7 +2,7 @@
   "author": "Lime Technologies",
   "main": "../dist/es5/testing/index.js",
-  "module": "../dist/testing/index.es.js",
+  "module": "../dist/testing/index.js",
```
👍 (I almost opened a PR to fix this myself, but didn't bother since we probably should stop using this undocumented field sometime anyway... https://stackoverflow.com/questions/42708484/what-is-the-module-package-json-field-for)

vicgeralds reviewed (commented):
Looks good, a few minor suggestions. I trust that the semanic release stuff just works 😄

jgroth commented:
I have rebased this and added two new commits. 
The first one changed the name of one of the services to be more consistent with the other ones (which is a breaking change)
The other one adds a missing interface for HTTP errors

jgroth commented:
I think I will remove 7efdd1b9 from this PR and merge it to `main` instead, it's been missing from there for a while as well and it could be good to introduce it before making a breaking change. 
I also noticed some other minor stuff that could be good to add before merging this while helping @Kiarokh with Lundalogik/lime-crm-components#1234

jgroth commented:
I found some more breaking changes we should get in before merging. Quite small changes, just some methods we had as optional previously
